src/librustc_ast/attr/mod.rs

```diff
@@ -590,6 +593,11 @@ impl MetaItemKind {
                 tokens.next();
                 MetaItemKind::list_from_tokens(inner_tokens)
             }
+            Some(TokenTree::Delimited(_, token::NoDelim, inner_tokens)) => {
```

Comment:
Have you seen this case in code? Could you add a test case?
I'm not sure how could `#[attr ⟪ 2 + 2 ⟫]` get through parser.

Reply:
This was needed to get libstd to compile - I'll try to minimize the case I found.

Reply:
The added test with `#[doc = $x]` corresponds to the other change above though.

Reply:
You're right - I've removed this match arm.